doc/manual/source/user_guide/InSituPythonAnalysis.rst

```diff
@@ -0,0 +1,269 @@
+
+``libyt`` supports calling Python functions from simulation process,
+and it also supports user interface (**interactive mode** has :ref:`Interactive Python Prompt` and :ref:`Reloading Script` feature, and **jupyter kernel mode** has :ref:`Jupyter Notebook / JupyterLab UI` feature).
+We can update Python functions and probe data interactively in the UI.
+After in situ analysis is done, ``libyt`` frees resources allocated for itself, and the simulation will continue.
+
+At the end, when the simulation is shutting down, Enzo calls ``FinalizeLibytInterface`` in ``src/enzo/InitializeLibytInterface.C``, so that ``libyt`` finalizes the Python interpreters.
+
+How to Configure
+----------------
+**Some settings are hard-coded inside Enzo, we can customize it to our own needs.**
+
+General
+^^^^^^^
+
+* **How to change import Python file name?**
+
+  The default Python script will be imported is ``inline.py``.
+  If we really want to change the name, we can go to
+  ``src/enzo/InitializeLibytInterface.C`` in function ``InitializeLibytByItself``, and change ``params->script`` to our Python file name without ``.py``. For example, I want to make it to ``test.py``:
+
+  ::
+
+      params->script = "test";
+
+  Please use ``const char*``, or else, we have to make sure the lifetime of this variable covers the whole in situ process.
+
+* **How to activate in situ Python analysis process?**
+
+  The full process is encapsulated inside ``CallInSitulibyt`` function in ``src/enzo/CallInSitulibyt.C``.
+  We can put this function everywhere we want in Enzo to start in situ analysis.
+  It will load and use Enzo's current state and data.
+
+  Currently, it is called inside ``EvolveLevel`` function.
+
+* **How to call Python functions during simulation runtime? And what should I be aware of?**
+
+  We can call Python function using libyt API ``yt_run_Function`` and ``yt_run_FunctionArguments``. See how to use them `here <https://libyt.readthedocs.io/en/latest/libyt-api/run-python-function.html>`__.
+
+  Put the API right after the comment
+
+  ::
+
+      // TODO: yt_run_Function and yt_run_FunctionArguments
+      // Put yt_run_Function and yt_run_FunctionArguments here
+
+  inside ``CallInSitulibyt`` function in ``src/enzo/CallInSitulibyt.C`` according to our needs.
+
+  Please make sure the functions we called are defined inside the script. Otherwise, in ``libyt`` normal modes, the simulation will terminate simply because it cannot find the Python function, while in the other modes, it will labeled as failed.
+
+  See how to use yt to do analysis `here <https://libyt.readthedocs.io/en/latest/in-situ-python-analysis/using-yt.html>`__.
+
+.. _Interactive Python Prompt:
+
+Interactive Python Prompt
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **How to activate interactive Python prompt in Enzo?**
+
+  We have to compile ``libyt`` in **interactive mode** and then un-comment the code in ``src/enzo/CallInSitulibyt.C``:
```

Comment:
Un-commenting code is cumbersome; maybe make an additional compiler configuration such as `libyt-interactive` similar to `libyt-yes` and `libyt-no`

Reply:
This is fixed.